### ISSUE TITLE
[BUG] npm v 8.11.0 - writes to STDERR - "npm WARN config global `--global`, `--local` ..." breaking our nightly build process.

### ISSUE THREAD

**Issue by raven1152** [npm/cli#4989]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

current version of npm (8.11.0, node 16.15.1) writes a warning to STDERR on every command, that is a warning about using deprecated options.  The message is...

``` npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead. ```

The message is written to STDERR regardless of whether or not the command run uses the --global or --local options.  For example, the command...

npm -v

outputs the message, even though -v is the only option used, and writes to STDERR, even though it states it's purpose to be a warning, not an error, and the command still executes successfully.

The result for my organization is to break our nightly builds.  Our builds need to monitor STDERR for REAL build errors, which also use STDERR to capture a variety of aberrant build conditions including security vulnerabilities that are reported back to the build pipeline via STDERR in a script that runs "npm audit" and analyzes the results.

![npm-warn](https://user-images.githubusercontent.com/18708621/171736746-431947f2-430d-4b29-b33f-c84284757d21.png)

### Expected Behavior

IMHO, this warning should only be written if the deprecated options are used, and should be written to STDOUT since it is not an error, just a warning.

### Steps To Reproduce

1. In Windows...
2. Running node 16.15.1 and npm 8.11.0
3. Run any npm command, i.e. npm -v
4. See results...
![npm output](https://user-images.githubusercontent.com/18708621/171734613-188a0d55-ad69-4b02-b7d0-26f1e15320d5.png)

### Environment

- npm: 8.11.0
- Node.js: 16.15.1
- OS Name: Windows 10
- System Model Name: Dell Precision 5500 series
- npm config:
```
; "user" config from C:\Users\xxxx\.npmrc

cafile = "C:\\cert\\OGS_int_CA.pem"

; node bin location = C:\Program Files\nodejs\node.exe
; node version = v16.15.1
; npm local prefix = C:\Users\xxxx
; npm version = 8.11.0
; cwd = C:\Users\xxxx
; HOME = C:\Users\xxxx
; Run `npm config ls -l` to show all defaults.
```

**Comment by ehoogeveen-medweb** [comment]
This issue seems to be coming from `Node.js` itself!

- With node `v16.15.1` I get this warning for every invocation of `npm`, even with npm `8.12.1` (which undeprecates `--global`)
- With node `v16.15.0` I don't get this warning, even with npm `8.11.0` (which deprecated `--global`)

I also only see the issue on Windows.

Edit: FWIW, the warning also triggers regardless of what is passed to npm - even explicitly passing `--location=global` or some other `--location` doesn't help. I think some dependency bundled with the release of Node.js v16.15.1 includes the deprecation warning, and updating the installed version of `npm` doesn't update this dependency (which may be a bug in and of itself?).

**Comment by XhmikosR** [comment]
This was fixed by #4982 and it's included in v8.12.1.

You should bug the Node.js people to update the bundled npm there if it's not already done and cut a new version. Alternatively, you can update your npm instance until the issue is fixed in Node.js.

**Comment by ehoogeveen-medweb** [comment]
Here's a workaround until Node.js v16.15.2 is released:

1. Open a Windows PowerShell command window with elevated privileges (an administrative prompt).
2. Run the following commands (assuming the default installation directory was used for Node.js):
```PowerShell
(Get-Content "${Env:ProgramFiles}\nodejs\npm.cmd") -replace '-g','-L=global' | Out-File -Encoding ascii "${Env:ProgramFiles}\nodejs\npm.cmd"
(Get-Content "${Env:ProgramFiles}\nodejs\npx.cmd") -replace '-g','-L=global' | Out-File -Encoding ascii "${Env:ProgramFiles}\nodejs\npx.cmd"
```

The problem here happens because the version of npm in the Node.js installation directory is used to get the global prefix. After that, processing is handed over to the global version of npm, but by that point the warning has already been printed. The version of npm in the Node.js installation directory does not get updated when you run `npm install -g npm@latest`.

Edit: Looks like this is still needed for Node.js v16.16.0!

**Comment by raven1152** [comment]
Thank you @ehoogeveen-medweb I have forwarded this to our build-master so they can implement the fix.  I appreciate your time in taking a look at this.

**Comment by tussar01** [comment]
Thanks

**Comment by K3NZ11** [comment]
I tried the [workaround](https://github.com/npm/cli/issues/4989#issuecomment-1146894284) from @ehoogeveen-medweb and it removed the warning only for `Powershell`. In my case, I was also using `GitBash`, and the `npm` command was executing from a different directory.

```PowerShell
# PowerShell -> C:\Program Files\nodejs\npm.cmd
Get-Command npm
```

```bash
# Git Bash -> C:\Program Files\nodejs\npm
which npm
```

Therefore, to apply this workaround for `GitBash`, you need to update the following files.
```bash
# Git Bash
sed -i 's/-g/-L=global/g' "C:/Program Files/nodejs/npm"
sed -i 's/-g/-L=global/g' "C:/Program Files/nodejs/npx"
```

**Comment by K3NZ11** [comment]
Upgrading Node.js to the [CURRENT version](https://nodejs.org/en/download/current/) that included npm `v8.15.0` might also remove this deprecated warning.
